src/functions.jl

```diff
@@ -655,6 +655,75 @@ end
 
 
 # Mutating functions for Taylor1{TaylorN{T}}
+@inline function zero!(a::Taylor1{T}, k::Int) where {T<:NumberNotSeries}
```

Comment:
Sorry for this late suggestion: what about redefining `zero!(c, a, k)` simply to point to `zero!(c, k)`? That would avoid lots of changes (which can be kept here), but in particular those related to `@taylorize` (in [dictmutfunct.jl](https://github.com/JuliaDiff/TaylorSeries.jl/blob/master/src/dictmutfunct.jl).

Reply:
Thanks for the suggestion! I changed the definition of `zero!(c, a, k)`

Reply:
One more suggestion on this: can you move the code (of the `zero!` methods introduced in this PR), closer to other `zero!` methods?